bluepysnap/nexus/connector.py

```diff
@@ -0,0 +1,200 @@
+# Copyright (c) 2022, EPFL/Blue Brain Project
+
+# This file is part of BlueBrain SNAP library <https://github.com/BlueBrain/snap>
+
+# This library is free software; you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License version 3.0 as published
+# by the Free Software Foundation.
+
+# This library is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+# details.
+
+# You should have received a copy of the GNU Lesser General Public License
+# along with this library; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+"""Implementation of Nexus connector for the kgforge API."""
+import logging
+from pathlib import Path
+
+L = logging.getLogger(__name__)
+
+NEXUS_NAMESPACE = "https://bluebrain.github.io/nexus/vocabulary/"
+PROJECTS_NAMESPACE = "https://bbp.epfl.ch/nexus/v1/projects/"
+USERS_NAMESPACE = "https://bbp.epfl.ch/nexus/v1/realms/bbp/users/"
+
+# Keys that need to be prefixed with the nexus namespace.
+# FIXME: should it be managed automatically in kgforge.core.archetypes.store.rewrite_sparql?
+_NEXUS_KEYS = {
+    "constrainedBy",
+    "createdAt",
+    "createdBy",
+    "deprecated",
+    "incoming",
+    "outgoing",
+    "project",
+    "schemaProject",
+    "rev",
+    "self",
+    "updatedAt",
+    "updatedBy",
+}
+
+_DATE_KEYS = {
+    "createdAt",
+    "updatedAt",
+}
+
+NAMESPACE_MAPPING = {
+    "createdBy": USERS_NAMESPACE,
+    "updatedBy": USERS_NAMESPACE,
+    "project": PROJECTS_NAMESPACE,
+}
+
+DATETIME_SUFFIX = "^^xsd:dateTime"
+
+
+def build_search_filters(type_, filters):
```

Comment:
is this internal to the module, or is it something that might be used by snap users?